### ISSUE TITLE
Raising on Repo.insert/update when changeset contains validation breaks constraints usage

### ISSUE THREAD

**Issue by simonprev** [jakub-zawislak/formex_ecto#7]
```elixir
    if errors != %{} do
      raise "Your changeset has errors. Since Formex 0.5, errors added in
      `Type.changeset_after_create_callback` are not being validated. You have to use
      new validator functionality. Errors found:\n
      "<>inspect(errors)
    end
```

This code breaks the usage of `formex_ecto` with `Ecto.Changeset.check_constraint/2` since it only validates after a DB roundtrip.

Why did this error was introduced in `formex_ecto`? I understand that it’s better to use `changeset_validation` since it is not tied to the DB. But constraint (like uniqueness) validation is a nice feature of Ecto that provides integrity of data at the database level.

**Comment by jakub-zawislak** [comment]
You are right. I forgot about that constraints stuff.

I just started working on it, and realised that this comment:
> errors added in `Type.changeset_after_create_callback` are not being validated

is not true, because errors are validated, but they are just not displayed 😄 So I need just to pass them back to the formex. 

I not finished fixing this issue yet because things are a bit complicated with collections, as always.

**Comment by simonprev** [comment]
Do you need help with this issue? Just a few pointers to get me started and I could start working on this 😄 

Will the new way of validating will allow us to use validations from a "changeset" validation function (like the classic Ecto way) instead of relying on validator per field?

**Comment by jakub-zawislak** [comment]
If you need this now, this code will work with forms without collections:

```elixir
  defp raise_changeset_error(form, changeset) do # <- I will change name of this function
    errors = changeset.errors
    |> Enum.reduce([], fn ({key, val}, acc) ->
      Keyword.update(acc, key, [val], &([val|&1]))
    end)

    form = Map.put(form, :errors, errors)
    form = Formex.Validator.translate_errors form

    {:error, %{form | submitted?: true}}
  end
```

You also have to change temporarily `Formex.Validator.translate_errors` function from private to public.

When I have form with collections, I see as many errors as many items in collections are :smile: I want to fix it this week.

I think that we will do not need `changeset_validation` callback anymore because the `changeset_after_create_callback` does almost the same. And I could change name of the latter to `modify_changeset`.

**Comment by jakub-zawislak** [comment]
I just published formex_ecto `0.2.0` and formex `0.6.6`. Please read [UPGRADE.md](https://github.com/jakub-zawislak/formex_ecto/blob/master/UPGRADE.md)

**Comment by simonprev** [comment]
This is awesome! Thank you very much @jakub-zawislak ! This will simplify a lot of our code 👍
